src/xhydro/extreme_value_analysis/parameterestimation.py

```diff
@@ -0,0 +1,526 @@
+        Supported distributions are genextreme, gumbel_r, genpareto.
+    method : {"ML", "PWM", "BAYES}
+        Fitting method, either maximum likelihood (ML), probability weighted moments (PWM) or bayesian (BAYES).
+        The PWM method is usually more robust to outliers.
+    dim : str
+        The dimension upon which to perform the indexing (default: "time").
+
+    Returns
+    -------
+    xr.DataArray
+        An array of fitted distribution parameters.
+
+    Notes
+    -----
+    Coordinates for which all values are NaNs will be dropped before fitting the distribution. If the array still
+    contains NaNs or has less valid values than the number of parameters for that distribution,
+    the distribution parameters will be returned as NaNs.
+    """
+    method = method.upper()
+    _check_fit_params(dist, method)
+    dist_params = _get_params(dist)
+    dist = get_dist(dist)
+
+    out = []
+    data = xr.apply_ufunc(
+        _fitfunc_1d,
+        ds,
+        input_core_dims=[[dim]],
+        output_core_dims=[["dparams"]],
+        vectorize=True,
+        dask="parallelized",
+        output_dtypes=[float],
+        keep_attrs=True,
+        kwargs=dict(
+            # Don't know how APP should be included, this works for now
+            dist=dist,
+            nparams=len(dist_params),
+            method=method,
+        ),
+        dask_gufunc_kwargs={"output_sizes": {"dparams": len(dist_params)}},
+    )
+    # Add coordinates for the distribution parameters and transpose to original shape (with dim -> dparams)
+    dims = [d if d != dim else "dparams" for d in ds.dims]
+    out = data.assign_coords(dparams=dist_params).transpose(*dims)
+
+    out.attrs = prefix_attrs(
+        ds.attrs, ["standard_name", "long_name", "units", "description"], "original_"
+    )
+    attrs = dict(
+        long_name=f"{dist.name} parameters",
+        description=f"Parameters of the {dist.name} distribution",
+        method=method,
+        estimator=METHOD_NAMES[method].capitalize(),
+        scipy_dist=dist.name,
+        units="",
+        history=update_history(
+            f"Estimate distribution parameters by {METHOD_NAMES[method]} method along dimension {dim}.",
+            new_name="fit",
+            data=ds,
+        ),
+    )
+    out.attrs.update(attrs)
+    return out
+
+
+def _fitfunc_1d(arr, *, dist, nparams, method):
+    """Fit distribution parameters."""
+    x = np.ma.masked_invalid(arr).compressed()  # pylint: disable=no-member
+    # Return NaNs if array is empty, which could happen at previous line if array only contained Nans
+    # if len(x) <= 1:
+    if len(x) <= nparams:  # TODO: sanity check with Jonathan
+        return np.asarray([np.nan] * nparams)
+
+    if method == "ML":
+        if dist == "genextreme" or str(type(dist)) == DIST_NAMES["genextreme"]:
+            # TODO: continue test for parametric quantile
+            param_list = gevfit(x)
```

Comment:
Si tu implémentes la logique avec `getattr` plus haut pour avoir juste une seule fonction `extremefit` (ou peu importe son nom), ça simplifiera cette section aussi.

Tu pourrais même construire le string on-the-fly en combinant `dist` et `method` --> `gumbel` et `PMW` --> `extremefit(x, "gumbelpmw")`. Faudra juste quelques `.replace()` pour `gev` et `ML`.